packages/cozy-client/src/CozyClient.js

```diff
@@ -477,20 +478,28 @@ export default class CozyClient {
   }
 
   async authorize(openURLCallback) {
-    const client = this.getClient()
-    const stateCode = client.generateStateCode()
-    const url = client.getAuthCodeURL(stateCode)
-
-    const redirectedURL = await openURLCallback(url)
-    const code = client.getAccessCodeFromURL(redirectedURL, stateCode)
-    const token = await client.fetchAccessToken(code)
-
-    client.setCredentials(token)
-
-    return {
-      token,
-      infos: client.oauthOptions,
-      client: client.oauthOptions // for compat with Authentication comp reasons
+    try {
+      const client = this.getClient()
+      const stateCode = client.generateStateCode()
+      const url = client.getAuthCodeURL(stateCode)
+      const redirectedURL = await openURLCallback(url)
+      const code = client.getAccessCodeFromURL(redirectedURL, stateCode)
+      const token = await client.fetchAccessToken(code)
+
+      client.setCredentials(token)
+      return {
+        token,
+        infos: client.oauthOptions,
+        client: client.oauthOptions // for compat with Authentication comp reasons
+      }
+    } catch (error) {
+      /* if REGISTRATION_ABORT is emited, we have to unregister the client. */
+      if (error.message === REGISTRATION_ABORT) {
+        const client = this.getClient()
+        client.unregister()
+        throw new Error(REGISTRATION_ABORT)
```

Comment:
this line is not mandatory ;)

Reply:
I worked on a cozy-error lib but didn't push it in this PR. 

The idea is to be able to throw some custom errors. With those customs errors, we can configure Sentry to not send it... Our (drive) Sentry is full of "non error", and this `Registration_abort` is part of it. 

I'll remove it for now.